packages/snaps-rpc-methods/src/restricted/notify.ts

```diff
@@ -200,5 +271,28 @@ export function getValidatedParams(params: unknown): NotifyParams {
     });
   }
 
-  return params as NotificationArgs;
+  try {
+    const validatedParams = createUnion(
+      params,
+      NotificationParametersStruct,
+      'type',
+    );
+
+    validateTextLinks(validatedParams.message, isOnPhishingList, getSnap);
+
+    if (hasProperty(validatedParams, 'footerLink')) {
+      validateTextLinks(
+        validatedParams.footerLink.text,
+        isOnPhishingList,
+        getSnap,
+      );
+      validateLink(validatedParams.footerLink.href, isOnPhishingList, getSnap);
+    }
+
+    return validatedParams;
+  } catch (error) {
+    throw rpcErrors.invalidParams({
+      message: `Invalid params: ${error.message}`,
```

Comment:
`error` isn't guaranteed to have a `message`. We have a `getErrorMessage` util function for this.